src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRepositoryClientImpl.java

```diff
@@ -54,16 +56,15 @@ public BitbucketWebhookClient getWebhookClient() {
     }
 
     @Override
-    public Stream<BitbucketPullRequest> getAllPullRequests() {
+    public Stream<BitbucketPullRequest> getPullRequests(@Nullable BitbucketPullRequestState state) {
```

Comment:
YAGNI right now, but if we ever need to make a "get all closed PRs" for example, this requires three invokations, which I think is the more likely case than wanting just declined PRs.

Reply:
yes, because of the Bitbucket API it will require either one (get all and filter yourself) or two (one for `declined` and one `merged`). I don't think the client should abstract that away. The BbS API is limited in this regard, it is either one state or no state you cannot pass in a list of states.